compiler/rustc_resolve/src/late.rs

```diff
@@ -3114,8 +3148,18 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
             if label.ident.as_str().as_bytes()[1] != b'_' {
                 self.diagnostic_metadata.unused_labels.insert(id, label.ident.span);
             }
+
+            let ident = label.ident.normalize_to_macro_rules();
+            for rib in self.label_ribs.iter_mut().rev() {
```

Comment:
Can this reuse `resolve_label` in some form instead?
It would fix all the issues with hygiene.

It would also more obviously followed the model "definition can resolved as a use -> report warning".